app/Command/Docs.hs

```diff
@@ -114,9 +110,19 @@ docgen (PSCDocsOptions fmt inputGlob output) = do
   takeByName' = takeModulesByName' D.modName
 
   parseAndConvert input =
-    runExceptT (D.parseFilesInPackages input []
-               >>= uncurry D.convertModulesInPackage)
-    >>= successOrExit
+    runExceptT (do
+      (modules, moduleMap) <- D.parseFilesInPackages input []
+
+      let moduleNameToFileMap =
+            M.fromList $ swap . fmap P.getModuleName <$> modules
+          getModuleFile docModule =
+            fromJust $ M.lookup (D.modName docModule) moduleNameToFileMap
```

Comment:
I’m not hugely keen on this. I guess this is to reassociate each file path with the relevant module after sending the modules through `convertModulesInPackage`? Is it worth considering changing `convertModulesInPackage` so that the output from `parseFilesInPackages` can just be passed straight in?

Reply:
I also find it a bit hackish and ugly. Happy to look at your proposal, and see if other options are better as well. Just a clarification: beyond the hackishness of this, is there anything else that you find unpleasant about it that I should be aware of?

Reply:
(and yes, your guess as to why this is done is correct)

Reply:
Makes sense. Thanks for the clarification.